portal/server/routes/contract/bond/index.js

```diff
@@ -189,16 +189,17 @@ const bondFinancialDetailsPayloadProperties = [
 ];
 
 const filterBondFinancialDetailsPayload = (body) => {
-  const payload = constructPayload(body, bondFinancialDetailsPayloadProperties);
-
-  if (payload.currencySameAsSupplyContractCurrency === 'true') {
-    delete payload.conversionRate;
-    delete payload['conversionRateDate-day'];
-    delete payload['conversionRateDate-month'];
-    delete payload['conversionRateDate-year'];
+  const propertyEmptyFlag = true; // Set to true if you want to validate empty properties
+  const sanitizedPayload = constructPayload(body, bondFinancialDetailsPayloadProperties, propertyEmptyFlag);
+
+  if (sanitizedPayload.currencySameAsSupplyContractCurrency === 'true') {
```

Comment:
```javascript
if (sanitizedPayload.currencySameAsSupplyContractCurrency) {
...
}
```

Reply:
Condition `if (sanitizedPayload.currencySameAsSupplyContractCurrency)` is not set to true then the condition is not evaluated.